e2e/testsuite/testsuite.go

```diff
@@ -126,62 +269,48 @@ func (s *E2ETestSuite) GetRelayerUsers(ctx context.Context, chainOpts ...ChainOp
 	return chainARelayerUser, chainBRelayerUser
 }
 
-// SetupChainsRelayerAndChannel create two chains, a relayer, establishes a connection and creates a channel
-// using the given channel options. The relayer returned by this function has not yet started. It should be started
-// with E2ETestSuite.StartRelayer if needed.
-// This should be called at the start of every test, unless fine grained control is required.
-func (s *E2ETestSuite) SetupChainsRelayerAndChannel(ctx context.Context, channelOpts func(*ibc.CreateChannelOptions), chainSpecOpts ...ChainOptionConfiguration) (ibc.Relayer, ibc.ChannelOutput) {
-	chains := s.GetAllChains(chainSpecOpts...)
-
-	r := relayer.New(s.T(), *LoadConfig().GetActiveRelayerConfig(), s.logger, s.DockerClient, s.network)
-
-	ic := s.newInterchain(ctx, r, chains, channelOpts)
-
-	buildOpts := interchaintest.InterchainBuildOptions{
-		TestName:  s.T().Name(),
-		Client:    s.DockerClient,
-		NetworkID: s.network,
-	}
-
-	s.Require().NoError(ic.Build(ctx, s.GetRelayerExecReporter(), buildOpts))
-
-	chainAChannels, err := r.GetChannels(ctx, s.GetRelayerExecReporter(), chains[0].Config().ChainID)
-	s.Require().NoError(err)
-	return r, chainAChannels[len(chainAChannels)-1]
-}
+// ChainOptionModifier is a function which accepts 2 chains as inputs, and returns a channel creation modifier function
+// in order to conditionally modify the channel options based on the chains being used.
+type ChainOptionModifier func(chainA, chainB ibc.Chain) func(options *ibc.CreateChannelOptions)
 
 // newInterchain constructs a new interchain instance that creates channels between the chains.
-func (s *E2ETestSuite) newInterchain(ctx context.Context, r ibc.Relayer, chains []ibc.Chain, channelOpts func(*ibc.CreateChannelOptions)) *interchaintest.Interchain {
-	channelOptions := defaultChannelOpts(chains)
-	if channelOpts != nil {
-		channelOpts(&channelOptions)
-	}
-
+func (s *E2ETestSuite) newInterchain(ctx context.Context, r ibc.Relayer, chains []ibc.Chain, modificationProvider ChainOptionModifier) *interchaintest.Interchain {
 	ic := interchaintest.NewInterchain()
 	for _, chain := range chains {
 		ic.AddChain(chain)
 	}
 	ic.AddRelayer(r, "r")
 
-	var pathNames []string
-
 	// iterate through all chains, and create links such that there is a channel between
 	// - chainA and chainB
 	// - chainB and chainC
 	// - chainC and chainD etc
 	for i := 0; i < len(chains)-1; i++ {
 		pathName := s.generatePathName()
-		pathNames = append(pathNames, pathName)
+		channelOpts := defaultChannelOpts(chains)
+		chain1, chain2 := chains[i], chains[i+1]
+
+		if modificationProvider != nil {
```

Comment:
provides a mechanism to selectively modify channels when there are more than 2 chains in a test.

Reply:
provide me modifications, almighty provider! 🙌🏻